src/Elcodi/Plugin/StoreSetupWizardBundle/EventListener/DashboardRedirectionEventListener.php

```diff
@@ -0,0 +1,118 @@
+<?php
+
+/*
+ * This file is part of the Elcodi package.
+ *
+ * Copyright (c) 2014 Elcodi.com
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ *
+ * Feel free to edit as you please, and have fun.
+ *
+ * @author Marc Morera <yuhu@mmoreram.com>
+ * @author Aldo Chiecchia <zimage@tiscali.it>
+ * @author Elcodi Team <tech@elcodi.com>
+ */
+
+namespace Elcodi\Plugin\StoreSetupWizardBundle\EventListener;
+
+use Symfony\Bundle\FrameworkBundle\Routing\Router;
+use Symfony\Component\HttpFoundation\RedirectResponse;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpKernel\Event\GetResponseEvent;
+
+use Elcodi\Component\Plugin\Entity\Plugin;
+use Elcodi\Plugin\StoreSetupWizardBundle\Services\WizardStatus;
+
+/**
+ * Class DashboardRedirectionEventListener
+ */
+class DashboardRedirectionEventListener
+{
+    /**
+     * @var Plugin
+     *
+     * Plugin
+     */
+    protected $plugin;
+
+    /**
+     * @var Router
+     *
+     * Router
+     */
+    protected $router;
+
+    /**
+     * @var WizardStatus
+     *
+     * A wizard status service
+     */
+    protected $wizardStatus;
+
+    /**
+     * Builds a new class
+     *
+     * @param Router       $router       A router
+     * @param WizardStatus $wizardStatus A wizard status service
+     */
+    public function __construct(
+        Router $router,
```

Comment:
Since you're only using `generator` method, better typehint for `UrlGeneratorInterface` instead of the full `Router`.